sycl/test/usm/findplatforms.hpp

```diff
@@ -0,0 +1,35 @@
+//==------------------- findplatforms.hpp ----------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+bool findPlatformAndDevice(cl_device_type deviceType,
+                           cl_platform_id &platformOut, cl_device_id &deviceOut,
+                           cl_int &errorCode) {
+  cl_uint numPlatforms;
+  bool foundDevice = false;
+
+  errorCode = clGetPlatformIDs(0, nullptr, &numPlatforms);
```

Comment:
Can we just return `false` if any of OpenCL API calls return value != `CL_SUCCESS`?
This function overrides `errorCode` value multiple time w/o checking intermediate results. Probably we can avoid using `errorCode` variable at all.

Reply:
Yes.

Reply:
How about checking `clGetPlatformIDs` error code?